### PR TITLE
Closes #78 | Create dataset loader for COCO-35L

### LINKED ISSUE #78: Create dataset loader for COCO-35L
Dataloader name: `coco_35l/coco_35l.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?coco_35l

|     Dataset| coco_35l |
|-------------|---|
| Description | COCO-35L is a machine-generated image caption dataset, constructed by translating COCO Captions (Chen et al., 2015) to the other 34 languages using Google’s machine translation API. |
| Subsets | fil, ind, tha, vie |
| Languages | fil, ind, tha, vie |
| Tasks | Image-to-Text Generation |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://google.github.io/crossmodal-3600/ |
| HF URL | - |
| Paper URL | https://aclanthology.org/2022.emnlp-main.45/ |

### PR DESCRIPTION
**Pull request by IvanHalimP**
Closes #78 
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by sabilmakbar** [comment]
Hi, I saw some confusing lines, also w/ several commented lines. Mind giving short comments on the scripts regarding the reasoning, clear commented lines (if they're not being used anymore), and reformat your dataloader `py` using `makefile` fomatter? Thx!

It'll give much better clarity on its codes and what you're trying to do with every lines, also ease the review process (and end-user understanding process of this dataloader).

File: seacrowd/sea_datasets/coco_35l/coco_35l.py
Comment (IvanHalimP) [inline seacrowd/sea_datasets/coco_35l/coco_35l.py:118]: Used to test the loader since I don't know which COCO version to be used. 2014 has more missing ```image_id```s compared to 2017, but the captions don't match. Thus it should be 2014. Will remove if everything is ok.

File: seacrowd/sea_datasets/coco_35l/coco_35l.py
Comment (IvanHalimP) [inline seacrowd/sea_datasets/coco_35l/coco_35l.py:127]: Used for debug, will remove later

File: seacrowd/sea_datasets/coco_35l/coco_35l.py
Comment (IvanHalimP) [inline seacrowd/sea_datasets/coco_35l/coco_35l.py:165]: Used for debug. Will remove later

**Comment by IvanHalimP** [comment]
> Hi, I saw some confusing lines, also w/ several commented lines. Mind giving short comments on the scripts regarding the reasoning, clear commented lines (if they're not being used anymore), and reformat your dataloader `py` using `makefile` fomatter? Thx!

image
I think this is very strange as I did run the make check before submitting it here. It also happens with my other PR if I'm not mistaken.

> It'll give much better clarity on its codes and what you're trying to do with every lines, also ease the review process (and end-user understanding process of this dataloader).

I'll add some description. Those comments are from my debugging, and I want to keep that just in case I need to make some modifications. Will be removed before merging. Just let me know.

**Comment by IvanHalimP** [comment]
> Hi, I saw some confusing lines, also w/ several commented lines. Mind giving short comments on the scripts regarding the reasoning, clear commented lines (if they're not being used anymore), and reformat your dataloader `py` using `makefile` fomatter? Thx!
image
Already did the make check before PR. Strangely, I got similar request frequently. Am I doing it wrong?


> It'll give much better clarity on its codes and what you're trying to do with every lines, also ease the review process (and end-user understanding process of this dataloader).

Will add brief explanations in the next commit. Those comments will be removed later. It is there just to keep track and make it easier for me to debug.

File: seacrowd/sea_datasets/coco_35l/coco_35l.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/coco_35l/coco_35l.py:72]: hmm kinda weird this one doesn't pass the formatted. I was referring to this line when talking abt formatter..

**Review by sabilmakbar — APPROVED** [review]
Update: It has been tested for all configs. LGTM for me

hi @holylovenia @SamuelCahyawijaya, since each config of dataloader requires around 20GB of downloaded data (per config, in total there are 4 config pairs -- amounting to 80GB), can it be considered for bonus points, per our last discussion?

I'm nominating for this as hard dataloader due to modality + data size + additional validation required (based on OP's comment)

**Comment by sabilmakbar** [comment]
Let's wait for Holy's review on this ya, @IvanHalimP

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @IvanHalimP, thanks a lot for the dataloader! Generally, it works fine and well, but I do have a minor change request. Could you please use `_LANGS = ["fil", "ind", "tha", "vie"]` instead of `_LANGS = ["fil", "id", "th", "vi"]`? Thanks again.

File: seacrowd/sea_datasets/coco_35l/coco_35l.py
Comment (holylovenia) [inline seacrowd/sea_datasets/coco_35l/coco_35l.py:82]: And could you please move this to the previous line?

**Comment by holylovenia** [comment]
> 152520 image ids are not found in the coco 2014 training caption. validation set is ok Using COCO 2014 train and validation set.

Could you please note this in the `_DESCRIPTION`?

**Comment by sabilmakbar** [comment]
Hi @IvanHalimP, have you got the time to check on the feedback based on @holylovenia's review?

**Comment by IvanHalimP** [comment]
Hi all, thank you for the comments. 
I nee to take care of something else for the time being, so please kindly wait. ;)

**Comment by holylovenia** [comment]
@IvanHalimP No problem, please take your time! Happy holidays!! ❄️

**Review by holylovenia — APPROVED** [review]

**Pull request merged**
